check.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+
+try_installing() {
+
+    if [ ! $NO_XBPS ]; then
+        echo "\nTrying to install '$PKG' ... "
+        if [ ! -z $SU_CMD ]; then
+            $SU_CMD xbps-install $PKG
+
+            # 'xbps-install' exits with 0 when the user aborts, so we can't
+            # check return status; use 'which' to check status instead.
+
+            if ( ! which mdbook 2>&- 1>&- ); then
+                echo "Can't proceed without '$PKG', aborting."
+                exit 1
+            fi
+        else
+            "Can't find 'sudo' or 'doas'; please install '$PKG' manually."
+            exit 1
+        fi
+    else
+        echo "Can't find 'xbps-install'; please install '$PKG' manually."
+        exit 1
+    fi
+
+}
+
+echo -n "Checking for mdbook ... "
+if ( ! which mdbook 2>/dev/null ); then
+    INSTALL_MDBOOK=1
+    echo "not found"
+fi
+
+echo -n "Checking for mdbook-linkcheck ... "
+if ( ! which mdbook-linkcheck 2>/dev/null ); then
+    INSTALL_LINKCHECK=1
+    echo "not found"
+fi
+
+echo -n "Checking for vmdfmt ... "
+if ( ! which vmdfmt 2>/dev/null ); then
+    INSTALL_VMDFMT=1
+    echo "not found"
+fi
+
+if ( ! which xbps-install 2>&- 1>&- ); then
+    NO_XBPS=1
+fi
+
+if ( which sudo 2>&- 1>&- ); then
+    SU_CMD='sudo'
+else
+    if ( which doas 2>&- 1>&- ); then
+        SU_CMD='doas'
+    else
+	SU_CMD=''
+    fi
+fi
+
+if [ $INSTALL_MDBOOK ]; then
+    PKG='mdBook'
+    try_installing
+fi
+
+if [ $INSTALL_LINKCHECK ]; then
+    PKG='mdbook-linkcheck'
+    try_installing
+fi
+
+if [ $INSTALL_VMDFMT ]; then
+    PKG='vmdfmt'
+    try_installing
+fi
```

Comment:
This one can end up running a `xbs-install` command for each package, right? Couldn't we put everything into a single command, and then in the end of `try_installing` check if all programs now exist?

Reply:
Sure, i'll see what i can do.